posthog/api/organization_feature_flag.py

```diff
@@ -96,10 +99,57 @@ def copy_flags(self, request, *args, **kwargs):
                 )
                 continue
 
-            context = {
-                "request": request,
-                "team_id": target_project_id,
-            }
+            # get all linked cohorts, sorted by creation order
+            seen_cohorts_cache = {}
+            sorted_cohort_ids = flag_to_copy.get_cohort_ids(
+                seen_cohorts_cache=seen_cohorts_cache, sort_by_creation_order=True
+            )
+
+            # destination cohort id is different from original cohort id - create mapping
+            name_to_dest_cohort_id = {}
+            # create cohorts in the destination project
+            if len(sorted_cohort_ids):
+                for cohort_id in sorted_cohort_ids:
+                    original_cohort = seen_cohorts_cache[str(cohort_id)]
+
+                    # search in destination project by name
+                    destination_cohort = Cohort.objects.filter(
+                        name=original_cohort.name, team_id=target_project_id, deleted=False
+                    ).first()
+
+                    # create new cohort in the destination project
+                    if not destination_cohort:
+                        new_filters = copy.deepcopy(original_cohort.filters)
+                        if new_filters:
+                            properties = new_filters.get("properties", [])
+                            for outer_prop in properties.get("values", []):
```

Comment:
this might be assuming too much about the data, since there's no guarantee that a property group will only have two levels, it can be recursively arbitrarily deep. I wonder if there's a better way to update this json 🤔  [separately, this requires it's own testing 😅 ]

Thinking about it.

Reply:
Something like this works much better because we leverage property group parsing to give us all props, and then operate on the props to generate what we need, and serialise that back for updating the cohort:


```
from posthog.models.filters.filter import Filter
from posthog.models.property import Property, PropertyGroup

data = {
    "type": "OR",
    "values": [
        {
            "type": "AND",
            "values": [
                {
                    "key": "email",
                    "value": "xyz",
                    "operator": "exact",
                    "type": "person",
                },
            ]
        },
        {
            "type": "OR",
            "values": [
                {
                    "type": "AND",
                    "values": [
                        {
                            "key": "email2",
                            "value": "xyz2",
                            "operator": "exact",
                            "type": "person",
                        },
                        {
                            "key": "email3",
                            "value": "xyz3",
                            "operator": "exact",
                            "type": "cohort",
                        },
                    ]
                },
                {
                    "type": "AND",
                    "values": [
                        {
                            "key": "email4",
                            "value": "xyz4",
                            "operator": "exact",
                            "type": "person",
                        },
                        {
                            "key": "email5",
                            "value": "xyz5",
                            "operator": "exact",
                            "type": "event",
                        },
                    ]
                },
            ]
        }
    ]
}

prop_group = Filter(data={"properties": data, "is_simplified": True}).property_groups

for prop in prop_group.flat:
    print(prop)
    print("\n")
    if prop.type == "person":
        prop.value = prop.value + "123"

print(prop_group.to_dict())
```

Just an example, I'm looking at person props instead and adding to the value, but idea is the same, the `to_dict()` in the end gives us the generated json which is correct for the cohort

Reply:
OOOH actually, this solves the backwards compatibility issue as well: never copy over cohort.groups, let them remain empty, and instead use `cohort.properties` (which generates a property group using both cohort.groups and any filters) to do the operation on, and save that as filters.

This effectively means you can replace

```
prop_group = Filter(data={"properties": data, "is_simplified": True}).property_groups
```

with

```
prop_group = original_cohort.properties
```

and it will handle everything 👍